pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -173,12 +173,18 @@ case 2: serving.kserve.org/deploymentMode is set
 	        if the mode is "RawDeployment", "Serverless" or "ModelMesh", return it.
 			else return config.deploy.defaultDeploymentMode
 */
-func GetDeploymentMode(annotations map[string]string, deployConfig *v1beta1.DeployConfig) constants.DeploymentModeType {
+func GetDeploymentMode(statusDeploymentMode string, annotations map[string]string, deployConfig *v1beta1.DeployConfig) constants.DeploymentModeType {
+	// First priority for deploymentMode is explicit request via annotation
 	deploymentMode, ok := annotations[constants.DeploymentMode]
 	if ok && (deploymentMode == string(constants.RawDeployment) || deploymentMode ==
 		string(constants.Serverless) || deploymentMode == string(constants.ModelMeshDeployment)) {
 		return constants.DeploymentModeType(deploymentMode)
 	}
+	// Second priority (to preserve original deploymentMode used when InferenceService was deployed)
```

Comment:
I wonder if the value in the `status` should be the first priority??

On PR description, you mention that by changing the default mode, KServe won't do cleanup. So, I may think that manually switching with the annotation from Serverless to Raw mode may also not clean the no longer needed resources for Serverless mode. If this is the case, it may be better to not allow switching the deployment mode (by honoring what was recorded in the `status`). KServe may later improve to do proper clean-up and the explicit annotation can have priority.

This would be my only comment. But, otherwise the approach here is looking good.

Reply:
agreed, changed the priority accordingly